server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -489,28 +489,48 @@ public void copyFrom(Directory from, String src, IOContext context, ActionListen
      * @param destinationPath The destination path (if multipart is supported)
      * @param fileCompletionListener The listener to notify of completion
      */
-    public void copyTo(String source, Directory destinationDirectory, Path destinationPath, ActionListener<String> fileCompletionListener) {
+    public void copyTo(
+        String source,
+        Directory destinationDirectory,
+        Path destinationPath,
+        DirectoryFileTransferTracker tracker,
+        ActionListener<String> fileCompletionListener
+    ) {
         final String blobName = getExistingRemoteFilename(source);
         if (destinationPath != null && remoteDataDirectory.getBlobContainer() instanceof AsyncMultiStreamBlobContainer) {
+            long length = 0L;
+            try {
+                length = fileLength(source);
```

Comment:
This makes a remote call to the repository to fetch the file length. I _think_ we actually already know the file length before making this call. Can we wire in the length instead of doing a remote call to fetch it?

Reply:
The method does that for us. It tries to use the cached value from the metadata map before making the network call. 
We can prevent it all together too though - that was the initial logic that we had, but this seemed better suited to report true stats.